network/src/application/storage.rs

```diff
@@ -1,18 +1,150 @@
 // Copyright (c) The Diem Core Contributors
 // SPDX-License-Identifier: Apache-2.0
-use crate::{
-    application::types::{PeerError, PeerInfo},
-    transport::ConnectionMetadata,
-};
-use diem_infallible::RwLock;
-use diem_types::PeerId;
+
 use std::{
     collections::{hash_map::Entry, HashMap},
     fmt::Debug,
     hash::Hash,
 };
 
-pub type PeerMetadataStorage = LockingHashMap<PeerId, PeerInfo>;
+use crate::{
+    application::types::{PeerError, PeerInfo},
+    transport::ConnectionMetadata,
+};
+use diem_config::network_id::{NetworkId, PeerNetworkId};
+use diem_infallible::{RwLock, RwLockWriteGuard};
+use diem_types::{account_address::AccountAddress, PeerId};
+use std::sync::Arc;
```

Comment:
this is why we combine all the imports, so cargo fmt can automatically merge this `std::sync::Arc` into the common `use std::{ .. }` block